### PR TITLE
add documentation on imagestore and add a warning if set

### LINKED ISSUE #1694: ImageStore and GraphRoot cannot be located in the same directory.
I believe this is a design decision but it'd be nice to have some documentation/warning on this.  

I configured my storage.conf to store everything in the same directory (and specified runroot, graphroot and imagestore) to be the same.

Once I did that, I was getting the following errors:

```
loading additional layer stores: lock ".." is not a read-only lock"
```

### PR DESCRIPTION
**Pull request by kannon92**
Fixes #1694 

During testing of imagestore in CRIO, I found this error.  Added some documentation and add an error if set to the same value.

### CODE REVIEW

File: storage.conf-freebsd [hunk 1]
```diff
@@ -22,6 +22,10 @@ runroot = "/var/run/containers/storage"
 # Primary Read/Write location of container storage
 graphroot = "/var/db/containers/storage"
 
+# Optional value for image storage location
+# If set must be different than graphroot.
+
+# imagestore =
```
Comment (flouthoc) [inline storage.conf-freebsd:28]:
```suggestion
# imagestore = ""
```

File: docs/containers-storage.conf.5.md [hunk 2]
```diff
@@ -56,10 +56,13 @@ $ restorecon -R -v /NEWSTORAGEPATH
 A common use case for this field is to provide a local storage directory when user home directories are NFS-mounted (podman does not support container storage over NFS).
 
 **imagestore**=""
+  image storage path (default is assumed to be same as `graphroot`).
```
[the remark this answers is not in the document]
Comment (kannon92) [reply docs/containers-storage.conf.5.md:59]: Yea I wanted to follow a similar pattern as other values.  The first sentence describes what the field is.

**Review by flouthoc — APPROVED** [review]
Few nits otherwise LGTM. @giuseppe @rhatdan @nalind PTAL

File: docs/containers-storage.conf.5.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kannon92) [reply docs/containers-storage.conf.5.md:59]: For example, graphroot:

```
graphroot="" container storage graph dir (default: "/var/lib/containers/storage")
```

File: storage.conf [hunk 3]
```diff
@@ -27,6 +27,11 @@ runroot = "/run/containers/storage"
 # restorecon -R -v /NEWSTORAGEPATH
 graphroot = "/var/lib/containers/storage"
 
+# Optional value for image storage location
+# If set must be different than graphroot.
+
+# imagestore =
```
[the remark this answers is not in the document]
Comment (kannon92) [reply storage.conf]: done.

File: storage.conf [hunk 4]
```diff
@@ -27,6 +27,11 @@ runroot = "/run/containers/storage"
 # restorecon -R -v /NEWSTORAGEPATH
 graphroot = "/var/lib/containers/storage"
 
+# Optional value for image storage location
+# If set must be different than graphroot.
```
Comment (TomSweeneyRedHat) [inline storage.conf:31]:
```suggestion
# If set, it must be different than graphroot.
```

File: storage.conf-freebsd [hunk 5]
```diff
@@ -22,6 +22,10 @@ runroot = "/var/run/containers/storage"
 # Primary Read/Write location of container storage
 graphroot = "/var/db/containers/storage"
 
+# Optional value for image storage location
+# If set must be different than graphroot.
```
Comment (TomSweeneyRedHat) [inline storage.conf-freebsd:26]: ditto prior

File: docs/containers-storage.conf.5.md (hunk 2 above)
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md:59]: soft suggestion for consideration
```suggestion
  image storage path (the default is assumed to be the same as `graphroot`).
```

**Review by TomSweeneyRedHat — COMMENTED** [review]
A couple of nits picked, otherwise LGTM

File: docs/containers-storage.conf.5.md (hunk 2 above)
Comment (kannon92) [reply docs/containers-storage.conf.5.md]: done.

File: storage.conf-freebsd (hunk 5 above)
Comment (kannon92) [reply storage.conf-freebsd]: done.

File: storage.conf (hunk 4 above)
Comment (kannon92) [reply storage.conf]: done.

File: types/options.go [hunk 6]
```diff
@@ -179,6 +179,10 @@ func defaultStoreOptionsIsolated(rootless bool, rootlessUID int, storageConf str
 		storageOpts.RootlessStoragePath = storagePath
 	}
 
+	if storageOpts.ImageStore != "" && storageOpts.ImageStore == storageOpts.GraphRoot {
+		return storageOpts, fmt.Errorf("imagestore if set must be a different location than graphroot")
```
Comment (rhatdan) [inline types/options.go:183]: It is set, o error should be.
return fmt..Errorf("imagestore %s must either be not set or be different then graphroot", storageOpts.ImageStore)

Comment (kannon92) [reply types/options.go]: updated.

**Comment by rhatdan** [comment]
LGTM
Thanks @kannon92

**Pull request merged**